### PR TITLE
Minor Improvements:

### PR DESCRIPTION
**Pull request by arturobernalg**
* Add final to variable
* Use 'Map.computeIfAbsent'

### CODE REVIEW

**Comment by garydgregory** [comment]
@arturobernalg 
May you please rebase?

**Comment by arturobernalg** [comment]
> @arturobernalg
> May you please rebase?

HI @garydgregory 

Done

File: src/main/java/org/apache/bcel/classfile/Utility.java [hunk 1]
```diff
@@ -831,7 +831,7 @@ public static String signatureToString( final String signature, final boolean ch
         // Could be Class or Type...
         type = typeSignatureToString(signature.substring(index), chopit);
         index += unwrap(CONSUMER_CHARS); // update position
-        if ((typeParams.isEmpty()) && (index == signature.length())) {
+        if ((typeParams.length() == 0) && (index == signature.length())) {
```
Comment (garydgregory) [inline src/main/java/org/apache/bcel/classfile/Utility.java:834]: -1: Don't in-line `isEmpty()` calls.

File: src/main/java/org/apache/bcel/classfile/Utility.java [hunk 2]
```diff
@@ -1014,71 +1014,71 @@ public static String typeSignatureToString( final String signature, final boolea
                     // we have TypeArguments; build up partial result
                     // as we recurse for each TypeArgument
                     final StringBuilder type = new StringBuilder(compactClassName(signature.substring(1, bracketIndex), chopit)).append("<");
-                    int consumed_chars = bracketIndex + 1; // Shadows global var
+                    int CONSUMER_CHARS = bracketIndex + 1; // Shadows global var
```
Comment (garydgregory) [inline src/main/java/org/apache/bcel/classfile/Utility.java:1017]: -1: don't upper case local vars.

**Review by garydgregory — CHANGES_REQUESTED** [review]

**Review by arturobernalg — COMMENTED** [review]
Changed

**Pull request closed without merging**